src/web/components/ClientSideTokenGeneration/CstgDomainsTable.tsx

```diff
@@ -132,18 +146,26 @@ export function CstgDomainsTable({
   const onSubmitAddDomainDialog = async (
     newDomainsFormatted: string[],
     deleteExistingList: boolean
-  ) => {
-    const newDomains = await onAddDomains(newDomainsFormatted, deleteExistingList);
-    setShowAddDomainsDialog(false);
-    setSearchedDomains(domains);
-    setSelectedDomains([]);
-    setSearchText('');
-    if (newDomains) {
-      setPageNumber(initialPageNumber);
-      setRowsPerPage(initialRowsPerPage);
-      setPagedDomains(getPagedDomains(newDomains, initialPageNumber, initialRowsPerPage));
-      setSearchedDomains(newDomains);
+  ): Promise<string[]> => {
+    const newDomainsResponse = await onAddDomains(newDomainsFormatted, deleteExistingList);
+    const newDomains = newDomainsResponse?.domains;
+    const isValidDomains = newDomainsResponse?.isValidDomains;
+    if (isValidDomains) {
```

Comment:
Perhaps the following would be slightly more readable?
```
if (isValidDomains) {
  setShowAddDomainsDialog(false);
  setSearchedDomains(domains);
  setSelectedDomains([]);
  setSearchText('');
  if (newDomains) {
    setPageNumber(initialPageNumber);
    setRowsPerPage(initialRowsPerPage);
    setPagedDomains(getPagedDomains(newDomains, initialPageNumber, initialRowsPerPage));
    setSearchedDomains(newDomains);
  }
  return [];
}
if (newDomains) {
  return newDomains;
}
return [];
```

Reply:
Better fix - thanks!